### ISSUE TITLE
page.apply_redactions gives unwanted black rectangle

### ISSUE THREAD

**Issue by CH-Tam** [pymupdf/PyMuPDF#3630]
### Description of the bug

I want to remove all texts and only keep vector graphics (such as straight lines) in PDF, the code and result are shown below.

However, the original PDF does not contain that black rectangle, the output is undesired. What should I do? (the PDF file is non-public, but it can be provided though private channels.)

Both 1.24.3 and 1.24.7 have this error.

### How to reproduce the bug

```
        doc = fitz.open(pdf_path)
                for page in doc:
                    page.add_redact_annot(page.rect + (-600, -600, 600, 600))
                    page.apply_redactions(graphics=0)
```

The result of 1st page is
![image](https://github.com/pymupdf/PyMuPDF/assets/52161076/894e69f4-a6e2-45fa-a7f4-f64464d0fab0)

### PyMuPDF version

1.24.7

### Operating system

Linux

### Python version

3.10

**Comment by JorjMcKie** [comment]
Please send me the file via e-mail, thanks.

**Comment by CH-Tam** [comment]
Hi,

test.pdf is the original PDF and result.pdf is the result. Page 1 & 2
contains unwanted black regions.

Much appreciated for your help.

Jorj X. McKie ***@***.***> 於 2024年6月27日 週四 下午2:12寫道：

**Comment by JorjMcKie** [comment]
@CH-Tam - I received nothing so far ... did you use jorj.x.mckie@outlook.de?

**Comment by JorjMcKie** [comment]
Thanks - file received.
This is a problem in the base library for which a fix has been developed.
I tested it locally with the improvements and confirmed that it works.

**Comment by JorjMcKie** [comment]
This has been fixed by a recent release - probably already 1.24.10.

**Comment by CH-Tam** [comment]
hi @JorjMcKie, I have tried the new versions (1.24.10 & 1.24.11). 
In 1.24.10, the issue is completely solved; however in 1.24.11, the above code has no impact at all (cannot clear the text in the PDF).
